fnet/src/vespa/fnet/transport_thread.cpp

```diff
@@ -556,10 +549,10 @@ FNET_TransportThread::endEventLoop() {
            _myQueue.IsEmpty_NoLock());
 
     {
-        std::lock_guard<std::mutex> guard(_lock);
+        std::lock_guard<std::mutex> guard(_shutdownLock);
         _finished = true;
         if (_waitFinished) {
-            _cond.notify_all();
+            _shutdownCond.notify_all();
```

Comment:
consider some extra cleanup by removing _waitFinished altogether and always notifying all.

Reply:
Godd point, I was wondering about its actual value.

Reply:
It is the 'avoid signals unless someone is waiting' pattern used elsewhere, but no real need for it here.